src/main/java/com/reshiftsecurity/plugins/intellij/service/EducationCachingService.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 Reshift Security Intellij plugin contributors
+ *
+ * This file is part of Reshift Security Intellij plugin.
+ *
+ * Reshift Security Intellij plugin is free software: you can redistribute it
+ * and/or modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * Reshift Security Intellij plugin is distributed in the hope that it will
+ * be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Reshift Security Intellij plugin.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.reshiftsecurity.plugins.intellij.service;
+
+import com.intellij.openapi.components.Service;
+import com.reshiftsecurity.education.DevContent;
+import com.reshiftsecurity.education.EducationService;
+import com.reshiftsecurity.education.VulnerabilityDetails;
+import com.reshiftsecurity.plugins.intellij.common.ExtendedProblemDescriptor;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class EducationCachingService {
+    private List<VulnerabilityDetails> contentCache;
+    private final String SECTION_OVERVIEW="Overview";
+
+    public EducationCachingService() {
+        this.contentCache = new ArrayList<>();
+    }
+
+    private Optional<VulnerabilityDetails> getContentFromCache(String issueType) {
+        for (VulnerabilityDetails vulnerabilityDetails : contentCache) {
```

Comment:
just a nitpick (can ignore): consider using the streaming api
contentCache.stream().filter(...).findAny().orElse(Optional.empty());